lib/std/os/windows.zig

```diff
@@ -255,7 +255,7 @@ pub fn CreatePipe(rd: *HANDLE, wr: *HANDLE, sattr: *const SECURITY_ATTRIBUTES) C
         GENERIC_READ | FILE_WRITE_ATTRIBUTES | SYNCHRONIZE,
         &attrs,
         &iosb,
-        FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
```

Comment:
How come you had to revert this change? Do you know why `CreateNamedPipeFile` is failing with your change?

Reply:
I guess because [`NtCreateNamedPipeFile` does not support `FILE_SHARE_DELETE`](https://github.com/tongzx/nt5src/blob/daad8a087a4e75422ec96b7911f1df4669989611/Source/XPSP1/NT/base/win32/client/pipe.c#L134).

Reply:
@kubkon

`CreatePipe()` was implemented recently in #19271 as (hopefully) an exact match of the same function in kernel32.
I think all changes to that function should be reverted. There is no extra issue here that could possibly arise with Zig's current flags that the vast majority of other software that just directly calls the kernel32 implementation also wouldn't have.

(The `\Device\Null` and `\??\MountPointManager` changes probably deserve a closer look too)

---

@chawyehsu

Please don't link to leaked source code without a warning (or at all).

Not everyone checks where the link leads before clicking on it, and even if they did, they might not recognize why it's a bad idea in this case.
Even just having this here could make it harder for people to contribute to Windows-related stuff in Zig, especially when it comes to working towards #1840 and similar issues.
ReactOS has had constant issues with this since its inception - I think we better err on the side of caution here. Thanks.